table/tables/tables.go

```diff
@@ -338,13 +340,23 @@ func (t *tableCommon) UpdateRecord(ctx sessionctx.Context, h int64, oldData, new
 		}
 	}
 	colSize := make(map[int64]int64)
+	encodedCol := make([]byte, 0, 16)
 	for id, col := range t.Cols() {
-		val := int64(len(newData[id].GetBytes()) - len(oldData[id].GetBytes()))
-		if val != 0 {
-			colSize[col.ID] = val
+		encodedCol = encodedCol[:0]
+		encodedCol, err = tablecodec.EncodeValue(sc, encodedCol, newData[id])
+		if err != nil {
+			continue
 		}
+		newLen := len(encodedCol) - 1
+		encodedCol = encodedCol[:0]
+		encodedCol, err = tablecodec.EncodeValue(sc, encodedCol, oldData[id])
```

Comment:
Would it impact the performance of the `update` statement?

Reply:
The bench result indicates that average time spent on this encoding for one column value in a row is 50ns on my laptop, which is cheaper than main memory reference according to https://serverfault.com/a/238534, though those numbers are 10 years old now.